### PR TITLE
Changed federated login

### PR DESCRIPTION
**Pull request by sveinpj**

### CODE REVIEW

**Review by oleterno-equinor — CHANGES_REQUESTED** [review]
Missing the `write-id` permission. 
Also update `deploy-prod-and-playground.yml` to use federated credentials.

**Review by oleterno-equinor — COMMENTED** [review]
Missing the permission for `deploy-dev.yml`.
Remove the commented code.

**Review by oleterno-equinor — COMMENTED** [review]
Change the `deploy-docker.yml` to only have one job, similar to [radix-acr-cleanup](https://github.com/equinor/radix-acr-cleanup/blob/master/.github/workflows/build-push.yml) and [radix-velero-plugin](https://github.com/equinor/radix-velero-plugin/blob/master/.github/workflows/push.yaml). 
Using `if`, specify to only push to radixprod ACR if the branch is `release`, as done in radix-velero-plugin.

File: .github/workflows/deploy-dev.yml [hunk 1]
```diff
@@ -18,11 +19,11 @@ jobs:
       ADMIN_GROUP_ID: "a5dfa635-dc00-4a28-9ad9-9e7f1e56919d"
       SQL_LOG_STORAGE_NAME: radixsqllogsdev
     steps:
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-dev.yml:21]: Needs the `actions/checkout@v2` step

File: .github/workflows/deploy-dev.yml [hunk 2]
```diff
@@ -18,11 +19,11 @@ jobs:
       ADMIN_GROUP_ID: "a5dfa635-dc00-4a28-9ad9-9e7f1e56919d"
       SQL_LOG_STORAGE_NAME: radixsqllogsdev
     steps:       
-      - uses: actions/checkout@v2
-      - name: Azure login
-        run: |
-          az login --service-principal --username ${{secrets.DEV_AZURE_CREDENTIALS_CLIENT_ID}}  --password "${{secrets.DEV_AZURE_CREDENTIALS_CLIENT_SECRET}}" --tenant ${{secrets.AZURE_TENANT_ID}}
-          az account set -s "${{secrets.DEV_AZURE_SUBSCRIPTION_ID}}"
+      - uses: azure/login@v1
+        with:
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-dev.yml:23]: Add a descriptive step name

File: .github/workflows/deploy-docker.yml [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
     steps:
-      - uses: actions/checkout@v2
+      - name: build-deploy-image-master
+        if: github.ref == 'refs/heads/master'
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID_DEV }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID_DEV }}
       - name: Generate build ID
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-docker.yml:23]: Move the `Generate build ID` step further up

File: .github/workflows/deploy-docker.yml [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
+  build:
+    name: Build-push
     env:
-      IMAGE: radix-cost-allocation
+      IMAGE: radix-velero-plugin
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - name: build-deploy-image-master
+        if: github.ref == 'refs/heads/master'
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-docker.yml:17]: This step should run for both master and release, so this `if` can be removed.

File: .github/workflows/deploy-docker.yml [hunk 5]
```diff
@@ -22,21 +30,18 @@ jobs:
...
-  build-deploy-image-release:
-    if: github.ref == 'refs/heads/release'
-    runs-on: ubuntu-latest
-    env:
-      IMAGE: radix-cost-allocation
-    steps:
+      - name: build-deploy-image-release
+        if: github.ref == 'refs/heads/release'
       - uses: actions/checkout@v2
       - name: Generate build ID
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-docker.yml:46]: The build ID only needs to be generated once. This step can be removed.

File: .github/workflows/deploy-docker.yml [hunk 6]
```diff
@@ -48,10 +53,11 @@ jobs:
       - name: Build Docker image dev
         run: |
           docker build -t radixdev.azurecr.io/${{ env.IMAGE }}:${{ steps.prep.outputs.BUILD_ID }} .
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-docker.yml:55]: At this point, the docker image has already been built for both dev and prod.

File: .github/workflows/deploy-docker.yml [hunk 7]
```diff
@@ -60,10 +66,11 @@ jobs:
       - name: Build Docker image prod
         run: |
           docker build -t radixprod.azurecr.io/${{ env.IMAGE }}:${{ steps.prep.outputs.BUILD_ID }} .
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-docker.yml:68]: At this point, the docker image has already been built for both dev and prod.

File: .github/workflows/deploy-docker.yml [hunk 8]
```diff
@@ -22,49 +24,36 @@ jobs:
...
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID_DEV }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID_DEV }}
+      - name: 'Login to ACR radixdev'
+        run: |
+          az acr login --name radixdev
       - name: Push to radixdev
         run: |
           az acr login -n radixdev -g common
```
Comment (oleterno-equinor) [inline .github/workflows/deploy-docker.yml:37]: Either do the ACR login in its own step or do it as part of another step. It should not be performed twice.

**Review by oleterno-equinor — APPROVED** [review]

File: .github/workflows/build-push.yml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID_PROD }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID_PROD }}
+      - name: 'Push to prod'
+        run: |
+          docker build -t radixprod.azurecr.io/${{ env.IMAGE }}:${{ steps.prep.outputs.BUILD_ID }} .
+          az acr login --name radixprod --resource-group common
+          docker push radixprod.azurecr.io/${{ env.IMAGE }}:${{ steps.prep.outputs.BUILD_ID }}
+          az logout
```
Comment (satr) [inline .github/workflows/build-push.yml:48]: This should be run only for release
`if: github.ref == 'refs/heads/release'`

**Review by satr — CHANGES_REQUESTED** [review]

**Pull request merged**